### PR TITLE
Handle malformed json better

### PR DESCRIPTION
**Pull request by luismfonseca**
The `JsonSyntaxException` was not being handled properly in most cases.

### CODE REVIEW

File: pushnotifications/src/main/java/com/pusher/pushnotifications/api/PushNotificationsAPI.kt [hunk 1]
```diff
@@ -36,6 +36,15 @@ class PushNotificationsAPI(private val instanceId: String) {
   var deviceId: String? = null
   var fcmToken: String? = null
 
+  // Handles the JsonSyntaxException properly
+  private fun safeExtractJsonError(possiblyJson: String): NOKResponse {
+    return try {
+      gson.fromJson(possiblyJson, NOKResponse::class.java)
+    } catch (jsonException: JsonSyntaxException) {
+      unknownNOKResponse
```
Comment (lukabratos) [inline pushnotifications/src/main/java/com/pusher/pushnotifications/api/PushNotificationsAPI.kt]: Would it be better if we would insert description of `JsonSyntaxException` into error message?

File: pushnotifications/src/main/java/com/pusher/pushnotifications/api/PushNotificationService.kt [hunk 2]
```diff
@@ -54,6 +54,8 @@ data class NOKResponse(
     get() = this.toString()
 }
 
+val unknownNOKResponse = NOKResponse("Unknown Service Error", "Something went wrong")
```
Comment (lukabratos) [inline pushnotifications/src/main/java/com/pusher/pushnotifications/api/PushNotificationService.kt]: Is there anything that we could do to make the error message more descriptive?

Comment (luismfonseca) [reply pushnotifications/src/main/java/com/pusher/pushnotifications/api/PushNotificationService.kt]: If we couldn't parse the json, then we don't know what went wrong. This is as descriptive as we can be.

Comment (lukabratos) [reply pushnotifications/src/main/java/com/pusher/pushnotifications/api/PushNotificationService.kt]: Should we say then in the error message that we were unable to parse the JSON?

File: pushnotifications/src/main/java/com/pusher/pushnotifications/api/PushNotificationsAPI.kt (hunk 1 above)
Comment (luismfonseca) [reply pushnotifications/src/main/java/com/pusher/pushnotifications/api/PushNotificationsAPI.kt]: I don't know. If we are not reporting back the exceptions, not sure if there's anything useful to do in here...

File: pushnotifications/src/main/java/com/pusher/pushnotifications/api/PushNotificationService.kt (hunk 2 above)
Comment (luismfonseca) [reply pushnotifications/src/main/java/com/pusher/pushnotifications/api/PushNotificationService.kt]: Unable to parse the JSON is a symptom of something going wrong, and not describing the underlying cause.

Something went wrong if this is unable to parse the error: the service was down? there was a bad proxy in the way? Some other thing we don't know about.

File: pushnotifications/src/main/java/com/pusher/pushnotifications/api/PushNotificationsAPI.kt (hunk 1 above)
Comment (luismfonseca) [reply pushnotifications/src/main/java/com/pusher/pushnotifications/api/PushNotificationsAPI.kt]: I'll add logging here.

**Pull request merged**